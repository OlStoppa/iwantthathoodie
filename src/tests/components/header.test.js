import React from "react"
import Header from "../../components/header"
import {render, cleanup } from "@testing-library/react"
import CartContext from "../../context/cartContext";

afterEach(cleanup)

it("renders", () => {

    const state = [[]]
    const {asFragment} = render(
        <CartContext.Provider value={state}>
        <Header />
        </CartContext.Provider>
        );
    expect(asFragment()).toMatchSnapshot();

});

test("renders the cart count", () => {
    const state =[[{qty: 1}, {qty: 3}]];
    const tree = (
        <CartContext.Provider value={state}>
        <Header />
        </CartContext.Provider>
    )

    const {getByTestId} = render(tree);
    expect(getByTestId('cart-counter').textContent).toBe('4');
})

