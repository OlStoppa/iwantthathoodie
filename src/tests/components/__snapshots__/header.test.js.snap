// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <header>
    <div
      style="margin: 0px auto; max-width: 1200px; padding: 0.5rem 1.45rem 0rem; display: flex; justify-content: space-between; align-items: center;"
    >
      <div
        class="menu-btn"
      >
        <div
          class="btn-line"
        />
        <div
          class="btn-line"
        />
        <div
          class="btn-line"
        />
      </div>
      <a
        href="/"
        style="color: white; text-decoration: none;"
      >
        <test-file-stub
          classname="logo"
        />
      </a>
      <a
        href="/cart"
      >
        <div
          class="cart-icon__container"
        >
          <span
            class="material-icons"
          >
            shopping_cart
          </span>
        </div>
      </a>
    </div>
    <nav
      class="navbar"
    >
      <ul>
        <a
          href="/collection/pullover"
        >
          <li>
            PULLOVERS
          </li>
        </a>
        <li>
          ZIP-UPS
        </li>
        <a
          href="/collection/t-shirt"
        >
          <li>
            T-SHIRTS
          </li>
        </a>
        <li>
          NEW
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
